### PR TITLE
Add unit test for AspNetCore instrumentation respecting suppress

### PR DESCRIPTION
**Pull request by utpilla**
Related to #1906 

## Changes
- Added a unit test AspNetCoreInstrumentationRespectSuppressInstrumentation to check if Sdk.SuppressInstrumentation is respected

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs [hunk 1]
```diff
@@ -496,6 +496,60 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Equal(shouldEnrichBeCalled, enrichCalled);
         }
 
+        [Fact]
```
Comment (reyang) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:499]: Consider making this parameterized:
1. suppress instrumentation = true, nothing gets called
2. suppress instrumentation = false, processor/propagator are called

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs [hunk 2]
```diff
@@ -496,6 +496,60 @@ void ConfigureTestServices(IServiceCollection services)
...
+                            this.openTelemetrySdk = Sdk.CreateTracerProviderBuilder()
+                                .AddAspNetCoreInstrumentation()
+                                .AddProcessor(activityProcessor.Object)
+                                .Build();
+                        })))
+                {
+                    using var scope = SuppressInstrumentationScope.Begin();
+                    using var client = testFactory.CreateClient();
+                    var response = await client.GetAsync("/api/values/2");
+                    response.EnsureSuccessStatusCode(); // Status Code 200-299
```
Comment (reyang) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:534]: Do we need this line?

Comment (utpilla) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:534]: It's not really needed, it's like being extra cautious to ensure that the request payload is not tampered with in anyway under SuppressInstrumentation.

Comment (reyang) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:534]: I see, makes sense!

**Review by reyang — APPROVED** [review]

**Pull request closed without merging**